### PR TITLE
Match SOM's toString and printing behaviour

### PR DESCRIPTION
**Pull request by ltratt**
Needs #43 to be merged first.

Previously I had relied on a hacky made-up primitive `println` to get *something* (anything!) printed from yksom before I had implementing things like inheritance etc. Those are now adequately enough implemented that we can follow SOM properly in the way it prints strings out.

This PR first implements SOM's printing primitives and definition of methods (https://github.com/softdevteam/yksom/commit/3c12fe4987015dfacc5b990f8c3a96de8effd670) before gradually removing yksom's temporary hack (https://github.com/softdevteam/yksom/commit/33d63d81bc9f3cb983d89627c7f2943dc55416a8 and https://github.com/softdevteam/yksom/commit/a94699062042df2a1a068ec38f45a8e94d0a5f20).

### CODE REVIEW

**Comment by ltratt** [comment]
Because of the changes to #43, it's probably best if I rebase this PR before review, so let me know if/when you would like me to do that.

**Comment by ltratt** [comment]
@ptersilie I suggest I rebase this before you review if that's OK?

File: src/lib/vm/vm.rs [hunk 1]
```diff
@@ -263,6 +269,16 @@ impl VM {
...
                 ValResult::from_val(rcv)
             }
+            Primitive::PrintNewline => {
+                println!();
+                ValResult::from_val(self.system.clone())
+            }
+            Primitive::PrintString => {
+                debug_assert_eq!(args.len(), 1);
+                let str_: &String_ = rtry!(args[0].downcast(self));
+                print!("{}", str_.as_str());
```
Comment (ptersilie) [inline src/lib/vm/vm.rs]: Forgot a debug print here? ;-)

File: src/lib/vm/vm.rs [hunk 2]
```diff
@@ -263,6 +269,16 @@ impl VM {
                 println!("{}", str_.as_str());
```
Comment (ptersilie) [inline src/lib/vm/vm.rs]: Whats up with that `XXX` on line 267? Anything we can do about that?

Comment (ptersilie) [reply src/lib/vm/vm.rs]: Nevermind. Just saw this is removed in the next commit.

**Comment by ptersilie** [comment]
Okay, let's rebase this! I'll delete the cache on travis so this should build fine after the rebase.

File: src/lib/vm/vm.rs (hunk 1 above)
Comment (ltratt) [reply src/lib/vm/vm.rs]: It's the print primitive, so it has to print something ;)

**Comment by ltratt** [comment]
Rebased.

**Comment by ptersilie** [comment]
Travis returned an error:
```
$ mv target/doc/yksom target/doc/api

mv: cannot move 'target/doc/yksom' to 'target/doc/api/yksom': Directory not empty

The command "mv target/doc/yksom target/doc/api" exited with 1.
```
Probably means we are caching the docs.

**Comment by ltratt** [comment]
OK, that worked. Should I squash?

**Comment by ptersilie** [comment]
Yes.

**Comment by ltratt** [comment]
Squashed.

**Pull request merged**